main.py

```diff
@@ -95,17 +180,43 @@ def draw() -> Shape:
         if events.type == generate_new_round and current_round <= 24:
             current_round = current_round + 1
 
-            if current_round > 24:
+            if current_round > 44:
+                data_management.database_commit()
                 draw_end_game()
-            else:
+            if current_round < 12:
                 clickable = True
                 winning_shape = draw()
                 last_shape = winning_shape.type
+                continue
+            if 12 < current_round < 24:
+                clickable = True
+                winning_shape = draw_second_stage()
+                last_shape = winning_shape.type
+                continue
 
         if events.type == pygame.MOUSEBUTTONDOWN and current_round <= 24 and clickable is True:
             clicked = SCREEN.get_at(pygame.mouse.get_pos())
             if clicked == winning_shape.colour:
                 clickable = False
                 print(f"You clicked the {winning_shape.type} successfully.")
-
+                if current_round == 1:
+                    stage = 'Baseline'
+                    data_management.create_timestamp(stage)
+
+                else:
+                    stage = f'Q{current_round}'
+                    data_management.create_timestamp(stage)
+        if 24 < current_round < 44:
+            if current_round not in whitespace_rounds:
+                file = draw_third_stage()
```

Comment:
This should use a different variable name since we have `file` already used in methods, so it is shadowed there.